### ISSUE TITLE
Best pattern for app-level permission checks?

### ISSUE THREAD

**Issue by SachaG** [meteor/guide#159]
A common pattern is redirecting to a different route if a user doesn't have the necessary rights to access a route. But this can't really be done at the router level, since it often requires waiting for subscriptions to load first. 

Is the guide going to cover the best way to do this? My current solution in Telescope is doing it in my layout template's helpers, but I'm not sure if that's a standard pattern or not.

**Comment by stubailo** [comment]
Yeah it's alluded to here: http://guide.meteor.com/routing.html#redirecting-dynamically

Basically, you've got it right. I thought we were going to cover that specific case here, but maybe it's up to me to do it in the users and accounts article. @tmeasday this used to be in routing, I thought - is it not anymore?

**Comment by tmeasday** [comment]
Doesn't that section exactly cover the use case? (I mean it's not auth-based, but it's the same problem of needing to wait on data before redirecting).

**Comment by SachaG** [comment]
OK, I hadn't seen this, sorry. Maybe we could also have an example that waits for route-specific data (for example, the current document) and not just global data?

**Comment by tmeasday** [comment]
Yeah, I guess a second example here would make sense.

**Comment by stubailo** [comment]
Yeah it would be good to have an example specifically for permissions maybe.

**Comment by tmeasday** [comment]
How's that?

**Comment by dburles** [comment]
I wonder whether people tend to jump on the idea of redirecting from experience with HTTP apps where this is the norm, but it's much better with Meteor.

For example rather than redirecting a user to a login template through the router, a much better pattern is likely to just use another layout on routes that require an authenticated user, perhaps `authenticatedLayout` which has some template logic:

```handlebars
{{#if currentUser}}
  {{Template.dynamic ...}}
{{else}}
  {{> loginForm}}
{{/if}}
```

You'll have to excuse me if that's already covered in the guide.

**Comment by stubailo** [comment]
I think we should cover it somewhere. Perhaps I should deal with this in the accounts guide? It doesn't really belong in routing or security.
